### PR TITLE
Fix the max pool kernel with channels_last memory layout

### LINKED ISSUE #109586: Max pool with negative integer inputs and channels_last memory layout gives the wrong values
### 🐛 Describe the bug

When running max pool operation on integral tensors with negative values and channels_last memory format the lowest value in the returned tensor is always zero.

``` python
import torch

t = torch.arange(24).view(1,1,4,6) - 12

expected = torch.max_pool2d(t, 2)
actual = torch.max_pool2d(t.clone(memory_format=torch.channels_last), 2)

assert torch.all(expected == actual).item()
# Note that the assert fails because
# expected is
# tensor([[[[-5, -3, -1],
#           [ 7,  9, 11]]]])
# actual is
#  tensor([[[[ 0,  0,  0],
#           [ 7,  9, 11]]]])
```

### Versions

Collecting environment information...
PyTorch version: 2.1.0.dev20230730+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: 14.0.6 (https://github.com/conda-forge/clangdev-feedstock ceeebe884c3cfd7160cf5a43e147f94439fafee3)
CMake version: version 3.25.3
Libc version: glibc-2.35

Python version: 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-79-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   48 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          60
On-line CPU(s) list:             0-59
Vendor ID:                       AuthenticAMD
Model name:                      AMD EPYC 7763 64-Core Processor
CPU family:                      25
Model:                           1
Thread(s) per core:              2
Core(s) per socket:              30
Socket(s):                       1
Stepping:                        1
BogoMIPS:                        4890.80
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext perfctr_core invpcid_single ssbd ibrs ibpb stibp vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr wbnoinvd arat npt nrip_save umip pku ospke vaes vpclmulqdq rdpid arch_capabilities
Virtualization:                  AMD-V
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       1.9 MiB (30 instances)
L1i cache:                       1.9 MiB (30 instances)
L2 cache:                        15 MiB (30 instances)
L3 cache:                        16 MiB (1 instance)
NUMA node(s):                    2
NUMA node0 CPU(s):               0-29
NUMA node1 CPU(s):               30-59
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-comprehensions==3.12.0
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] flake8-simplify==0.19.3
[pip3] mypy==1.4.1
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.1
[pip3] torch==2.1.0.dev20230730+cpu
[pip3] torchvision==0.16.0.dev20230730+cpu
[pip3] triton==2.0.0
[conda] numpy                     1.25.2          py310ha4c1d20_0    conda-forge
[conda] torch                     2.1.0.dev20230730+cpu          pypi_0    pypi
[conda] torchvision               0.16.0.dev20230730+cpu          pypi_0    pypi

### PR DESCRIPTION
**Pull request by Anstow**
Fixes #109586

In the kernel for the `channels_last` memory layout pytorch doesn't set the initial value correctly for integer inputs (setting the it to zero rather than the actual minimum value). This causes us to get the wrong value out for max pool if all the inputs are negative.

This fixes the onnx test (for int8) because the onnx test checks that the value returned by onnx is the same as the value pytorch computes and it is now computed correctly in pytorch.

ps: This is not related to ONNX. An ONNX test was fixed as the result of fixing a pytorch kernel

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: test/onnx/test_fx_op_consistency.py
Comment (thiagocrepaldi) [inline test/onnx/test_fx_op_consistency.py:394]: why `torch.int8` was removed (by expanding `onnx_test_common.INT_TYPES`)?

File: test/onnx/test_fx_op_consistency.py
Comment (thiagocrepaldi) [inline test/onnx/test_fx_op_consistency.py:399]: same here, but also for bools

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]
There is no issue linked nor description to the PR

Please describe what was the issue and how it was fixed and why the ONNX tests were affected by this change

**Comment by Anstow** [comment]
Hey is anyone looking at this?

File: test/onnx/test_fx_op_consistency.py
Comment (Anstow) [reply test/onnx/test_fx_op_consistency.py:394]: Because it now passes the tests. I didn't dig in to why. I imagine it's because I've fixed our implementation

File: test/onnx/test_fx_op_consistency.py
Comment (Anstow) [reply test/onnx/test_fx_op_consistency.py:399]: Same because it now passes the tests

Comment (Anstow) [reply test/onnx/test_fx_op_consistency.py:399]: There aren't any bool types invocations for max_pool*d which is why it doesn't matter if it's there or not

**Review by thiagocrepaldi — COMMENTED** [review]
Changes for `test/onnx/test_fx_op_consistency.py` are OK.

You still need someone's from pytorch core to review the maxpool change.

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by Anstow** [comment]
How do I ping the core team to take a look?

**Comment by thiagocrepaldi** [comment]
I have removed the `triaged` label. Let's see if that helps.

**Comment by Anstow** [comment]
OK removing the triaged label appears not to have helped

**Review by sanchitintel — COMMENTED** [review]
LGTM, thanks for the fix!
I'm not the code-owner of this code, though, so an approval from my account wouldn't help towards landing.

@jgong5 is away this week, so we can either wait for this PR to be triaged, so that another reviewer would be assigned, or we can wait until @jgong5 can help review the PR. Thanks!

**Review by jgong5 — APPROVED** [review]

**Comment by sanchitintel** [comment]
The macOS CI failure looks legit.

**Review by kulinseth — APPROVED** [review]

**Comment by Anstow** [comment]
Yeah, I don't have access to a mac so I'm having to iterate on CI. I've re-disabled the max pool 3d tests (and re-enabled the low precision max pol 2d tests that I'd accidentally disabled)

**Pull request closed without merging**
